libcoap/src/message/request.rs

```diff
@@ -570,9 +427,12 @@ impl CoapRequest {
 
     /// Converts this request into a [CoapMessage] that can be sent over a [CoapSession](crate::session::CoapSession).
     pub fn into_message(mut self) -> CoapMessage {
-        if let Some(req_uri) = self.uri {
-            req_uri.into_options().into_iter().for_each(|v| self.pdu.add_option(v));
+        if self.uri.is_proxy() {
+            self.pdu.add_option(CoapOption::ProxyUri(
+                self.uri.scheme().expect("Parsed CoAP URI must have scheme").to_string(),
+            ))
         }
+        self.uri.into_options().into_iter().for_each(|v| self.pdu.add_option(v));
```

Comment:
Can this be simplified to the following?
```suggestion
        self.uri.into_options().into_iter().for_each(self.pdu.add_option);
```

Reply:
Okay, I think I just forgot that this isn't possible in Rust, my bad